pkg/database/database.go

```diff
@@ -75,7 +81,20 @@ func ConnectDB(
 			}
 		}
 
-		s, err := sql.Open(sqlite.DriverName, strings.Replace(dbURL.String(), "sqlite://", "file:", 1))
+		sql.Register(SQLiteCustomDriverName, &sqlite3.SQLiteDriver{
+			ConnectHook: func(conn *sqlite3.SQLiteConn) error {
+				return conn.RegisterFunc("regexp", func(re, s string) bool {
+					c, err := regexp.Compile(re)
```

Comment:
please add an LRU cache of compiled regexes — I did a super basic benchmark and cut the time in half by adding a super basic `map[string]*regexp.Regexp` cache

Reply:
do you think we need to take care about concurrency?

Reply:
yes, definitely!

Reply:
ook.

Reply:
done.